src/lib/storage/local/local-storage.js

```diff
@@ -296,7 +301,12 @@ class LocalStorage {
 
       data.versions[version] = metadata;
       this.utils.tag_version(data, version, tag);
-      this.localList.add(name);
+
+      let addFailed = this.localList.add(name);
+      if (addFailed) {
+        return cb(this.utils.ErrorCode.get422(addFailed.message));
```

Comment:
We will need a functional test for this, I can help on that. But you will need to merge in a branch instead of the master branch.